src/main/scala/com/databricks/labs/overwatch/ApiCallV2.scala

```diff
@@ -0,0 +1,515 @@
+              .postData(jsonQuery)
+              .options(reqOptions)
+              .asString
+          } catch {
+            case _: javax.net.ssl.SSLHandshakeException => // for PVC with ssl errors
+              val sslMSG = "ALERT: DROPPING BACK TO UNSAFE SSL: SSL handshake errors were detected, allowing unsafe " +
+                "ssl. If this is unexpected behavior, validate your ssl certs."
+              logger.log(Level.WARN, sslMSG)
+              if (debugFlag) println(sslMSG)
+              if (unsafeSSLErrorCount == 0) { //Check for 1st occurrence of SSL Handshake error.
+                unsafeSSLErrorCount += 1
+                allowUnsafeSSL = true
+                Http(s"""${apiEnv.workspaceURL}/${apiMeta.apiV}/${endPoint}""")
+                  .copy(headers = httpHeaders)
+                  .postData(jsonQuery)
+                  .options(reqOptions)
+                  .asString
+              } else {
+                throw new Exception(sslMSG)
+              }
+
+
+          }
+      }
+      case "GET" => {
+        response = try {
+          Http(s"""${apiEnv.workspaceURL}/${apiMeta.apiV}/${endPoint}""")
+            .param(jsonKey, jsonValue)
+            .copy(headers = httpHeaders)
+            .options(reqOptions)
+            .asString
+        } catch {
+          case e: javax.net.ssl.SSLHandshakeException if !allowUnsafeSSL => // for PVC with ssl errors
+            val sslMSG = "ALERT: DROPPING BACK TO UNSAFE SSL: SSL handshake errors were detected, allowing unsafe " +
+              "ssl. If this is unexpected behavior, validate your ssl certs."
+            logger.log(Level.WARN, sslMSG)
+            if (debugFlag) println(sslMSG)
+            if (unsafeSSLErrorCount == 0) { //Check for 1st occurrence of SSL Handshake error.
+              unsafeSSLErrorCount += 1
+              allowUnsafeSSL = true
+              Http(s"""${apiEnv.workspaceURL}/${apiMeta.apiV}/${endPoint}""")
+                .param(jsonKey, jsonValue)
+                .copy(headers = httpHeaders)
+                .options(reqOptions)
+                .asString
+            } else {
+              throw new Exception(sslMSG)
+            }
+          case e:Exception=> throw e
+        }
+      }
+
+
+    }
+    response
+  }
+
+
+  /**
+   * Creating a generic error message
+   * @return
+   */
+  private def buildGenericErrorMessage: String = {
+    s"""API CALL FAILED: Endpoint: ${endPoint}
+       |jsonQuery:${jsonQuery}
+       |""".stripMargin
+  }
+
+
+  /**
+   * Get the required columns from the received dataframe.
+   * @param rawDF The raw dataframe which is received as part of API response.
+   * @return Dataframe which contains the required columns.
+   */
+  def processRawData(rawDF: DataFrame): DataFrame = {
+    val resultDFFieldNames = rawDF.schema.fieldNames
+    if (!resultDFFieldNames.contains(apiMeta.dataframeColumns) && apiMeta.dataframeColumns != "*") { // if known API but return column doesn't exist
```

Comment:
suggest changing name from `dataframeColumns` to `dataframeColumn` since it's a string and only accepts a single column. If more than one column were to be passed in map logic and an array would need to be used. Unnecessary for now but maybe needed in future. Wording indicates that this may already be possible.